sdks/python/apache_beam/io/textio.py

```diff
@@ -86,7 +90,22 @@ def __init__(self,
                coder,
                buffer_size=DEFAULT_READ_BUFFER_SIZE,
                validate=True,
-               skip_header_lines=0):
+               skip_header_lines=0,
+               header_processor_fns=(None, None)):
+    """Initialize a _TextSource
+
+    Args:
+      header_processor_fns (tuple): a tuple of a header_matcher function
+        and a header_processor function. The header_matcher should return True
+        for all lines at the start of the file that are part of the file header
+        and False otherwise. These header_lines will be skipped over when
+        reading records and instead passed into header_processor to be handled.
```

Comment:
And then what is the role of `header_processor`? Also, why not put this in two separate arguments?

Reply:
header_lines will be passed to the header_processor function. By "These header_lines will be skipped over when reading records", I meant that the _read_records function will not yield these lines. I'll change the description to make this a bit more clear.